mirror.py

```diff
@@ -18,72 +17,64 @@ class MirrorUploader(object):
     IMPLEMENTATION_REGISTRY = {}
 
     @classmethod
-    def sitewide(cls, _db):
-        """Create a MirrorUploader from a sitewide configuration.
+    def mirror(cls, _db, storage_name=None, integration=None):
+        """Create a MirrorUploader from an integration or storage name.
+
+        :param storage_name: The name of the S3 storage integration.
+        :param integration: The S3 external integration.
```

Comment:
It's good to have a docstring here but we're doing this to eventually support MirrorUploaders other than S3, so let's not start with the assumption that this is an S3 integration specifically.